src/api/auth/LighthouseAuthApi.ts

```diff
@@ -76,19 +118,41 @@ export class LighthouseAuthApi implements AuthApi {
       method: 'POST',
       credentials: 'include',
     });
-
     return true;
   }
 
   async getPublicUsers(): Promise<User[]> {
-    throw new Error('Method not implemented.');
+    // TODO: we currently don't have the concept of public users in the new API (heimdall)
+    return this.getAllUsers();
   }
 
   async getAllUsers(): Promise<User[]> {
-    throw new Error('Method not implemented.');
+    const apiUsersResponse = await fetch(`${this.url}/users`, {
+      credentials: 'include',
+    });
+    const apiUsers: ApiUser[] = await apiUsersResponse.json()
+    const users: User[] = apiUsers.map(apiUser => {
+      const user: User = {
+        username: apiUser.username,
+        role: undefined, // TODO: change role to roles
+        course: apiUser.registration_key?.key,
+        createdAt: new Date(apiUser.created_at),
+        lastSeen: new Date(apiUser.last_login),
+      };
+      return user;
+    });
+    return users;
   }
 
   async getToken(): Promise<Token | null> {
-    throw new Error('Method not implemented.');
+    const apiTokenResponse = await fetch(`${this.url}/users/${this.user?.id}/api-token`, {
```

Comment:
We should add a check to avoid accidentally interpolating `null`/`undefined` here:

```suggestion
    if (!this.user) {
      return null;
    }
    const apiTokenResponse = await fetch(`${this.url}/users/${this.user.id}/api-token`, {
```

Maybe something more fancy.

Reply:
Good catch :+1: